### ISSUE TITLE
Allow indentation of non-nested code

### ISSUE THREAD

**Issue by arcanis** [causes/scss-lint#233]
There should be a setting allowing to indent non-nested code. For example, the following is valid and is often used to tell the reader that there is a kind of hierarchy, but without using fifteen kilometers long selectors :

```scss
.foo {
}

    .something-inside-foo {
    }

        .and-something-inside-its-parent {
        }
```

Instead of

```scss
.foo {
    .something-inside-foo {
        .and-something-inside-its-parent {
        }
    }
}
```

Doing this generates

```scss
.foo { }
.something-inside-foo { }
.and-something-inside-its-parent { }
```

Instead of

```scss
.foo { }
.foo .something-inside-foo { }
.foo .something-inside-foo .and-something-inside-its-parent { }
...
```

I suggest adding an option to the Indentation linter to ignore indentation errors when they increase the indentation level of a rule by a single level, and only if the rule is not the first one of the block.

**Comment by sds** [comment]
Implemented in 1b0f2e5b140cd2f7ac145a93901eee6296e93529.

**Comment by ArmorDarks** [comment]
Great, thanks a lot!

**Comment by kty** [comment]
This doesn't support class names with parent selectors.
As of Sass 3.3 parent selectors (`&`) can be used for prefixing class names, which means we can now rewrite this:

```scss
.block {}
.block__table {}
.block__row {}
```

into this: 

```scss
.block {
    &__table {}
    &__row {}
}
```

Let's say `&__row` will always be inside `&__table` and I'd like to suggest that in the code:

```scss
.block {
    &__table {}
        &__row {}
}
```

This triggers a lint error, although being (IMO) a legitimate use of arbitrary indentation.

Would you consider removing the condition that arbitrary indentation is allowed only for top-level rulesets? Thanks!

**Comment by sds** [comment]
Hey @kty, we would happily accept a pull request adding support for the feature you describe. Thanks!
